From c9224878cba7a1c164e1d657a53e7338f834b468 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Wed, 30 Sep 2020 16:44:39 +0800
Subject: [PATCH 17/68] gst-play: export custom gstplay api

add GST_PLAY_API declare to expose custom api

Upstream-Status: Pending
Signed-off-by: Haihua Hu <jared.hu@nxp.com>

(cherry picked from commit 6cc15dd52acbfa02af00656249879086a27dee11)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/play/gstplay.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gst-libs/gst/play/gstplay.h b/gst-libs/gst/play/gstplay.h
index 2c53aeb..831e4c9 100644
--- a/gst-libs/gst/play/gstplay.h
+++ b/gst-libs/gst/play/gstplay.h
@@ -445,21 +445,33 @@ GST_PLAY_API
 void           gst_play_message_parse_muted_changed              (GstMessage *msg, gboolean *muted);
 
 /* Custom gstplay API */
+GST_PLAY_API
 gboolean    gst_play_set_rotate (GstPlay * play, gint rotation);
+GST_PLAY_API
 gint        gst_play_get_rotate (GstPlay * play);
 
+GST_PLAY_API
 void        gst_play_config_set_force_aspect_ratio (GstPlay * self, gboolean force_aspect_ratio);
+GST_PLAY_API
 gboolean    gst_play_config_get_force_aspect_ratio (const GstStructure * config);
 
+GST_PLAY_API
 gboolean    gst_play_set_audio_sink (GstPlay * play, GstElement * audio_sink);
+GST_PLAY_API
 gboolean    gst_play_set_text_sink (GstPlay * play, GstElement * text_sink);
+GST_PLAY_API
 GstElement * gst_play_get_audio_sink (GstPlay * play);
+GST_PLAY_API
 GstElement * gst_play_get_text_sink (GstPlay * play);
 
+GST_PLAY_API
 GstPlayState  gst_play_get_state (GstPlay * play);
 
+GST_PLAY_API
 void         gst_play_play_sync (GstPlay * play, gint time_out);
+GST_PLAY_API
 void         gst_play_stop_sync (GstPlay * play, gint time_out);
+GST_PLAY_API
 void         gst_play_pause_sync (GstPlay * play, gint time_out);
 
 G_END_DECLS
-- 
2.43.0

